src/common/io.cc

```diff
@@ -7,6 +7,11 @@
 #include <sys/mman.h>  // for mmap, mmap64, munmap, madvise
 #include <unistd.h>    // for close, getpagesize
 
+#include "xgboost/unixdefs.h"
+#ifdef __MUSL__
+#define mmap64 mmap
```

Comment:
We can use `mmap` as default and let the 32-bit system raise an overflow error when the file size is larger than 2GB.